extensions/src/main/java/org/pentaho/platform/web/http/api/resources/RepositoryResource.java

```diff
@@ -650,6 +654,23 @@ private boolean hasOperationId( final List<IPluginOperation> operations, final S
 
   protected Response doService( String contextId, String resourceId ) throws ObjectFactoryException,
       PluginBeanException, IOException, URISyntaxException {
+    // Check if we are editing a content
+    String perspectiveId = resourceId;
+    if ( perspectiveId != null && perspectiveId.indexOf( "." ) >= 0 ) {
+      String[] parts = perspectiveId.split( "\\." );
+      if( parts != null && parts.length > 0) {
+        perspectiveId = parts[1];
+      }
+    }
+
+    if( perspectiveId != null && ( perspectiveId.equals( "editor" ) || perspectiveId.equals( "edit" ) ) ) {
+      // Check if user has permission to edit the content. If they do not have access, throw and error
+      if( !canEdit() ) {
+        logger.error( Messages.getInstance().getString( "RepositoryResource.USER_NOT_AUTHORIZED_TO_EDIT" ) );
+        return buildSafeHtmlServerErrorResponse( Messages.getInstance().getString( "RepositoryResource.USER_NOT_AUTHORIZED_TO_EDIT" ) );
+      }
+    }
```

Comment:
Please see my comment https://github.com/pentaho/pentaho-platform/pull/5131#discussion_r963114501 regarding the location where this check is currently being performed not being the "correct" one.